src/main/java/org/folio/rest/impl/PermsAPI.java

```diff
@@ -1713,195 +1267,126 @@ public void getPermsPermissions(String expandSubs, String expanded, String inclu
-      Context vertxContext, String tenantId) {
+  private Future<Void> removeSubpermissionFromPermissionList(Conn connection, String subpermissionName, List<String> permissionNameList) {
 
     if (permissionNameList.isEmpty()) {
       return Future.succeededFuture();
     }
     List<String> permissionNameListCopy = new ArrayList<>(permissionNameList);
     String permissionName = permissionNameListCopy.get(0);
     permissionNameListCopy.remove(0);
-    Promise<Void> promise = Promise.promise();
-    removeSubpermissionFromPermission(connection, subpermissionName, permissionName,
-        vertxContext, tenantId).onComplete(rsfpRes -> promise.handle(rsfpRes.mapEmpty()));
-    return promise.future().compose(res -> removeSubpermissionFromPermissionList(connection, subpermissionName,
-        permissionNameListCopy, vertxContext, tenantId));
+    return removeSubpermissionFromPermission(connection, subpermissionName, permissionName)
+        .compose(res -> removeSubpermissionFromPermissionList(connection, subpermissionName,
+            permissionNameListCopy));
   }
 
-  private Future<Void> removeSubpermissionFromPermission(
-      AsyncResult<SQLConnection> connection, String subpermissionName, String permissionName,
-      Context vertxContext, String tenantId) {
+  private Future<Void> removeSubpermissionFromPermission(Conn connection, String subpermissionName, String permissionName) {
 
-    Promise<Void> promise = Promise.promise();
-    try {
-      Criteria nameCrit = new Criteria()
-          .addField(PERMISSION_NAME_FIELD)
-          .setOperation("=")
-          .setVal(permissionName);
-      Criterion criterion = new Criterion(nameCrit);
-      CQLWrapper cqlFilter = new CQLWrapper(criterion);
-      PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(),
-          tenantId);
-      pgClient.get(connection, TABLE_NAME_PERMS, Permission.class,
-          criterion, true, false, getReply -> {
-            if (getReply.failed()) {
-              promise.fail(getReply.cause());
-              return;
-            }
-            if (getReply.result().getResults().isEmpty()) {
-              promise.complete();
-              return;
-            }
-            Permission permission = getReply.result().getResults().get(0);
-            permission.getSubPermissions().remove(subpermissionName);
-              pgClient.update(connection, TABLE_NAME_PERMS, permission, cqlFilter,
-                  true, updateReply -> promise.handle(updateReply.mapEmpty()));
-          });
-    } catch (Exception e) {
-      promise.fail(e);
-    }
-    return promise.future();
+    Criteria nameCrit = new Criteria()
+        .addField(PERMISSION_NAME_FIELD)
+        .setOperation("=")
+        .setVal(permissionName);
+    Criterion criterion = new Criterion(nameCrit);
+    CQLWrapper cqlFilter = new CQLWrapper(criterion);
+    return connection.get(TABLE_NAME_PERMS, Permission.class,
+        criterion, false)
```

Comment:
Leaving out the `false` also doesn't calculate totalRecords.

Reply:
yep